src/openrct2/ride/Vehicle.cpp

```diff
@@ -1781,14 +1781,21 @@ static void vehicle_update_measurements(rct_vehicle* vehicle)
             ride->start_drop_height = vehicle->z / 8;
         }
 
-        if (track_flags & TRACK_ELEM_FLAG_NORMAL_TO_INVERSION)
+        if (ride->type == RIDE_TYPE_MINI_GOLF)
         {
-            uint8_t inversions = ride->inversions & 0x1F;
-            if (inversions != 0x1F)
-                inversions++;
-
-            ride->inversions &= ~0x1F;
-            ride->inversions |= inversions;
+            if (track_flags & TRACK_ELEM_FLAG_IS_GOLF_HOLE)
+            {
+                if (ride->holes < 255)
```

Comment:
This brakes compatibility with vanilla. Stick to 0x1F

Reply:
How does it break compatibility? It gets &'ed with 0x1F when exporting.

Reply:
How does it break compatibility? It gets truncated on export. I suppose I could use a `min(31, inversions)` on export rather than truncating.

Reply:
`min(31, inversions)` would be better.

Reply:
Done.

Reply:
You have to change it here to 31 otherwise you modify state when saving. This will break multiplayer and also mean the save performs differently in vanilla and open

Reply:
How will it break multiplayer, apart from needing a network version bump?

Really, I don't care at all that vanilla behaves differently for the few months we still export to it: it only happens with more than 31 inversions/holes, and even then it is handled gracefully: RCT2 wouldn't know the difference.

Reply:
The checksums wont match between clients and hosts as the files are sent as S6's.

Reply:
May as well keep it compatible for the next few months before increasing the limits.